crates/chia-tools/src/bin/benchmark-condition-costs.rs

```diff
@@ -344,18 +349,21 @@ pub fn main() {
         //     puz_hash_node_ptr,
         //     Arc::new(coin.coin_id()),
         // );
-        // Create the conditions
-        let conditions = create_conditions(&mut allocator, &cond).expect("create_conditions");
-        // a "spend" is the following list (parent puzhash amount conditions)
-        let spend_list = [parent_id, puz_hash_node_ptr, hundred, conditions];
-        let mut spends = allocator.nil();
-        for arg in spend_list.iter().rev() {
-            spends = allocator.new_pair(*arg, spends).expect("new_pair");
-        }
-        let mut cost = TEST_CONSTANTS.max_block_cost_clvm;
+        let mut cost = u64::MAX;
         let mut samples = Vec::<(f64, f64)>::new();
+        let mut signature = Signature::default();
+        let cp = allocator.checkpoint();
         // Parse the conditions and then make the list longer
-        for i in 0..1000 {
+        let mut conditions =
+            create_conditions(&mut allocator, &cond, 1).expect("create_conditions");
+        for i in 1..500 {
+            signature += &cond.aggregate_signature;
+            let mut spends = allocator.nil();
+            // a "spend" is the following list (parent puzhash amount conditions)
+            let spend_list = [parent_id, puz_hash_node_ptr, hundred, conditions];
+            for arg in spend_list.iter().rev() {
+                spends = allocator.new_pair(*arg, spends).expect("new_pair");
+            }
```

Comment:
I think you should use clvm_traits here instead

Reply:
I think it's worth not using clvm_traits so we can reuse the atom pointers